flatcar-linux/kubernetes/butane/controller.yaml

```diff
@@ -117,27 +121,70 @@ systemd:
         Environment=KUBELET_IMAGE=quay.io/poseidon/kubelet:v1.26.3
         ExecStart=/usr/bin/docker run \
             -v /etc/kubernetes/pki:/etc/kubernetes/pki:ro \
-            -v /opt/bootstrap/assets:/assets:ro \
+            -v /${opt}/bootstrap/assets:/assets:ro \
             -v /opt/bootstrap/apply:/apply:ro \
             --entrypoint=/apply \
             $${KUBELET_IMAGE}
         ExecStartPost=/bin/touch /opt/bootstrap/bootstrap.done
         [Install]
         WantedBy=multi-user.target
+    %{ if ! enable_install }
+    - name: update-engine.service
+      mask: true
+    - name: mnt-persist.mount
+      enabled: true
+      contents: |
+        [Unit]
+        Description= Mounting persistent data
+        After=persistent_preparation.service
+        DefaultDependencies=no
+        [Mount]
+        What=/dev/disk/by-label/persistent
+        Where=/mnt/persist
+        Type=ext4
+        Options=defaults
+        [Install]
+        WantedBy=multi-user.target
+    - name: persistent_preparation.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description= A service used to prepare the persistent disk by creating the required partition if it doesn't exist
+        [Service]
+        Type= oneshot
+        ExecStart=/opt/create_persistent_partition
+        RemainAfterExit= false
+        [Install]
+        WantedBy=multi-user.target
```

Comment:
I thought I would have issues with the `device` part, since what I'm doing here is either using the persist_disk variable if it is specified or choosing the smallest disk available on the machine. 

if we consider `persist_disk` a mandatory variable, I think we can easily switch to this one

Reply:
done